### PR TITLE
Comment: add link to the OSTI.GOV URL for the optimizing document

### PR DESCRIPTION
**Pull request by DougBurke**
# Summary

Add a reference to the OSTI.GOV technical report in the comments in the C++ code used for testing the optimization functions.

# Details

@dtnguyen2 mentioned this in a PR or issue and I keep on finding it hard to find, so adding it to the actual documentation is a good idea.

### CODE REVIEW

File: sherpa/optmethods/tests/tstoptfct.hh [hunk 1]
```diff
@@ -27,6 +27,8 @@
 // "Algorithm 566: Fortran Subroutines for Testing Unconstrained
 // Optimization Software.", ACM TOMS, VOL. 7, PAGES 14-41 AND 136-140, 1981
 //
+// See also https://www.osti.gov/biblio/6650344
+//
 // The comments on the solutions were taken from the site:
 //
 // http://www.uni-graz.at/imawww/kuntsevich/solvopt/results/moreset.html
```
Comment (hamogu) [inline sherpa/optmethods/tests/tstoptfct.hh:34]: @dtnguyen2 This link does not exist any longer... Do you happen to have a copy? If not, shall we remove it?

Comment (DougBurke) [reply sherpa/optmethods/tests/tstoptfct.hh:34]: Oh, you're talkinga bout the uni-gratz URL. I thought for a second you were talking about the link I just added!

Comment (hamogu) [reply sherpa/optmethods/tests/tstoptfct.hh:34]: @DougBurke Your link is fine and likely to be around for a while. It just shows link rot, in particular on personal websites.

So, I check the wayback machine and will add a new link.

Comment (hamogu) [inline sherpa/optmethods/tests/tstoptfct.hh:34]:
```suggestion
// See also https://www.osti.gov/biblio/6650344
//
// The comments on the solutions were taken from the site:
//
// https://web.archive.org/web/20060221201940/http://www.uni-graz.at/imawww/kuntsevich/solvopt/results/moreset.html
```

Comment (DougBurke) [reply sherpa/optmethods/tests/tstoptfct.hh]: I've added the link @hamogu suggested but I'll leave this comment open for now

**Review by dtnguyen2 — APPROVED** [review]

**Pull request merged**